sklearn/utils/estimator_checks.py

```diff
@@ -1191,6 +1192,56 @@ def check_estimators_pickle(name, estimator_orig):
         assert_allclose_dense_sparse(result[method], unpickled_result)
 
 
+def check_pairwise_estimator_tag(name, estimator_orig):
+    attributes_to_check = ['metric', 'affinity', 'kernel']
+
+    # Check if _pairwise attribute is present - will be used later
+    has_pairwise_tag = (False
+                        if getattr(estimator_orig, '_pairwise', None) is None
+                        else True)
+
+    # Using iris as sample data
+    iris = load_iris()
+    X, y_ = iris.data, iris.target
+    distance_matrix = pairwise_distances(X)
+
+    for attribute in attributes_to_check:
+        # Check to see if attribute value is supported by estimator
+        if attribute not in estimator_orig.get_params(deep=False):
+            continue
+        try:
+            # Construct new object of estimator with desired attribute value
+            modified_estimator = clone(estimator_orig).set_params(
+                                                **{attribute: 'precomputed'})
+            # Estimators may validate parameters in fit if not in set_params
+            modified_estimator.fit(distance_matrix, y_)
+        except Exception:
+            # Estimator does not support given attribute value
+            continue
+
+        # Also check to see if non-square distance matrix raises an error
+        try:
+            non_square_distance = distance_matrix[:, :-1]
+            if getattr(modified_estimator, 'fit_predict', None) is not None:
+                modified_estimator.fit_predict(non_square_distance, y_)
+            elif (getattr(modified_estimator, 'fit_transform', None)
+                    is not None):
+                modified_estimator.fit_transform(non_square_distance, y_)
+            elif getattr(modified_estimator, 'fit', None) is not None:
+                modified_estimator.fit(non_square_distance, y_)
+                if getattr(modified_estimator, 'predict', None) is not None:
+                    modified_estimator.predict(X)
```

Comment:
`modified_estimator` everywhere is cumbersome. Better to use `estimator2` or something similar.

Reply:
renamed `modified_estimator` to `estimator2`